src/main/java/com/liveramp/captain/example/ExampleCaptainWorkflow.java

```diff
@@ -0,0 +1,349 @@
+package com.liveramp.captain.example;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.liveramp.captain.app_type.CaptainAppType;
+import com.liveramp.captain.daemon.CaptainBuilder;
+import com.liveramp.captain.daemon.CaptainConfigProducer;
+import com.liveramp.captain.daemon.CaptainRequestConfig;
+import com.liveramp.captain.daemon.RequestUpdater;
+import com.liveramp.captain.daemon.SimpleCaptainConfig;
+import com.liveramp.captain.example.internals.MockAnalyticsService;
+import com.liveramp.captain.example.internals.MockDataScience;
+import com.liveramp.captain.example.internals.MockDb;
+import com.liveramp.captain.example.internals.MockIngestionService;
+import com.liveramp.captain.example.internals.MockRequest;
+import com.liveramp.captain.handle_persistor.HandlePersistor;
+import com.liveramp.captain.manifest.DefaultManifestImpl;
+import com.liveramp.captain.manifest.Manifest;
+import com.liveramp.captain.manifest_manager.ManifestManager;
+import com.liveramp.captain.manifest_manager.MultiAppManifestManager;
+import com.liveramp.captain.notifier.CaptainNotifier;
+import com.liveramp.captain.request_context.RequestContext;
+import com.liveramp.captain.request_submitter.NoOpSubmitter;
+import com.liveramp.captain.request_submitter.RequestSubmitter;
+import com.liveramp.captain.status.CaptainStatus;
+import com.liveramp.captain.status_retriever.StatusRetriever;
+import com.liveramp.captain.step.CaptainStep;
+import com.liveramp.captain.waypoint.AsyncWaypoint;
+import com.liveramp.captain.waypoint.ControlFlowWaypoint;
+import com.liveramp.captain.waypoint.SyncWaypoint;
+import com.liveramp.commons.collections.map.MapBuilder;
+import com.liveramp.daemon_lib.Daemon;
+
+public class ExampleCaptainWorkflow {
```

Comment:
General thoughts and kind've just thought about it now.

I think it'd be very useful to allow users to see the number of interactions called. I think I had a bit of trouble understanding at first when I was reading through Captain code, how often my config producer was being called, how often the waypoint code is being called (like checking status), etc.

Do you think it'd be useful to have certain classes in this toy example have a static counter that just gets increased each time it gets called and then expose some way to see that?  Not sure if thats totally reasonable or easy to do or even worth it.

Reply:
yeah, i think that's a really great idea. not sure how to do it though.

Reply:
talked offline. we agreed this should be in the v2 of the example app.